From 760b784c78b247defc1c13e42d5aa0d9bb2aa021 Mon Sep 17 00:00:00 2001
From: Maruthi Srinivas Bayyavarapu <maruthi.srinivas.bayyavarapu@xilinx.com>
Date: Thu, 20 Sep 2018 14:01:09 +0530
Subject: [PATCH 0118/1851] ASoC: xlnx: Use graph API instead of phandle

commit a27118c2818e0fd884da1c0bf9e16bad57350593 from
https://github.com/Xilinx/linux-xlnx.git

SDI embed node now uses graph endpoint binding instead of phandle to
SDI Tx video node. Changed corresponding API to get the video node.

Signed-off-by: Maruthi Srinivas Bayyavarapu <maruthi.srinivas.bayyavarapu@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 sound/soc/xilinx/xlnx_sdi_audio.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/sound/soc/xilinx/xlnx_sdi_audio.c b/sound/soc/xilinx/xlnx_sdi_audio.c
index d8e4ff244f96..a3bba4aca5cc 100644
--- a/sound/soc/xilinx/xlnx_sdi_audio.c
+++ b/sound/soc/xilinx/xlnx_sdi_audio.c
@@ -9,6 +9,7 @@
 #include <linux/interrupt.h>
 #include <linux/io.h>
 #include <linux/module.h>
+#include <linux/of_graph.h>
 #include <linux/of_platform.h>
 #include <linux/platform_device.h>
 #include <linux/wait.h>
@@ -667,8 +668,8 @@ static int xlnx_sdi_audio_probe(struct platform_device *pdev)
 		snd_dai = &xlnx_sdi_rx_dai;
 	} else {
 		ctx->mode = EMBED;
-		video_node = of_parse_phandle(pdev->dev.of_node,
-					      "xlnx,sdi-tx-video", 0);
+
+		video_node = of_graph_get_remote_node(pdev->dev.of_node, 0, 0);
 		if (!video_node) {
 			dev_err(ctx->dev, "video_node not found\n");
 			of_node_put(video_node);
-- 
2.31.1

